Improve appearance of quick reply send button

fezjo opened the pull request:
Fixes #1933 

Improve appearance of send button in conversation compose box. Improvements include:
- support for dark mode
- cursor and color change on hover (old design didn't react in any way)
- more padding and boxier design more in line with modern Thunderbird style

Comparison:

Current Light
![before-light](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/3646376d-ced6-4bc2-ba53-a77d137f7102)

Current Dark
![before-dark](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/48b3b3df-e181-4bfa-884e-2c2f6c6b84f3)

New Light
![after-light-normal](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/6c10a40b-7edf-416d-9f2c-453b39a7461f)
![after-light-hover](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/f4a50b02-e37a-461f-8094-1a3643a5ad93)

New Dark
![after-dark-normal](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/3ffbbc6f-1e37-4dd3-8727-c0c8e118cf5e)
![after-dark-hover](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/1328b519-03b2-4429-8c6c-33f17f985e96)

It seems like the surrounding box doesn't resize properly. The bottom edge of the button is close to the border and _maybe_ on some other machines it will stick out. I am not really well versed in HTML and CSS so I don't know how to fix the layout properly. However, this change as it is is still an improvement.

Future improvements should also include changing links `discard`, `Forward this conversation`, `Print this conversation` to button style. It seems like the UI has been neglected and can be easily improved.

Standard8 reviewed (changes requested):
Thank you for trying this, sorry for the delay in getting back to you.

As you indicated, the bottom border isn't quite right, and I'm not sure I'd be happy accepting it as-is.

I think what it needs is:

* The `margin-top` changing to `padding-top` [on `.compose > buttons`](https://github.com/thunderbird-conversations/thunderbird-conversations/blob/8b177a0d0fce22c1cd54fa656088aa597e8715d0/addon/content/components/compose/composeWidget.css#L66-L68).
* Removing the [`float: right` on `#send`](https://github.com/thunderbird-conversations/thunderbird-conversations/blob/8b177a0d0fce22c1cd54fa656088aa597e8715d0/addon/content/components/compose/composeWidget.css#L79C1-L81) and replacing with something that still gets the button aligned right.
  * This is needed as otherwise the float is causing the button to overlap padding when it is rendered without float.
  * I think you could do this with the following added to `.compose > .buttons`:

```
display: flex;
justify-content: space-between;
```

I think then it'd give a consistent spacing around the button.

Standard8 commented:
> Future improvements should also include changing links `discard`, `Forward this conversation`, `Print this conversation` to button style. It seems like the UI has been neglected and can be easily improved.

Changing the discard to a button would be reasonable. I don't think I'd agree with changing the forward/print to buttons, I think that would not fit correctly with the general layout/style.

fezjo commented:
Hello, your suggestions worked flawlessly (as far as I can tell). Thanks!

I also changed the discard link to button and changed the tests accordingly.

Finally, I added pictograms in the form of emojis next to the buttons so it's clear at a glance what the actions does. This is again just a suggestion, and may not be the best solution. Firstly, the trash can may be misleading as the email is not moved to _Trash_ but discarded. Also, I am not sure if all clients/fonts support these emojis and Thunderbird uses SVG icons. I would also prefer to use the same icons as Thunderbird, but this seemed like a quick and trivial initial improvement. Again, if you don't like the style or have suggestions, feel free to say or change anything.

![image](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/102b9956-4247-4e39-a17e-da1c0f6947a7)

Standard8 commented:
For the icons, if you rebase on the latest, you'll see that I've just added a `send` icon to `material-icons.svg`. There's already a `delete_forever` icon, that I think we could use for the discard.

In theory, you should be able to use the [`SvgIcon` class to display the icons](https://github.com/thunderbird-conversations/thunderbird-conversations/blob/8b177a0d0fce22c1cd54fa656088aa597e8715d0/addon/content/components/message/messageActionButton.jsx#L76).

fezjo commented:
Here's the version with SVG icons. I put the send icon to the right because it looked weird on the left and now it forms a big arrow with the text.

Also, I put a `nbsp` in between. As you can maybe tell, I don't really know what I am doing with web stuff :D

![image](https://github.com/thunderbird-conversations/thunderbird-conversations/assets/26350212/40e61427-3afa-4bfc-8ce1-a166f8e7eb51)

Standard8 commented on addon/content/components/compose/composeWidget.css:
```diff
@@ -64,18 +64,19 @@ input:focus {
-  background-color: white;
+  padding: 0.3em 1em;
+  background-color: -moz-dialog;
+  color: -moz-dialogtext;
```
This seems fine in dark mode (though maybe a little bright?), but in light mode it is too harsh compared to the rest of the UI.

Normally in light mode we're using `#666`: https://github.com/thunderbird-conversations/thunderbird-conversations/blob/1f5a213c734bafc535587a6d65cf54166865a4f4/addon/content/conversation.css#L152

So I think we want to have `#666` as the default and `-moz-dialogtext` as the `(prefers-color-scheme: dark)` option.

fezjo replied:
Well if that's what you normally do, then of course use 666.

However, maybe all of your UI should eventually be consistent with the user theme. But that's work for other PR...

Standard8 replied:
Yeah, let's be consistent for now, and we can explore different options in later options.

Standard8 commented on addon/content/components/compose/composeWidget.jsx:
```diff
@@ -99,11 +100,11 @@ export function ComposeWidget({ discard }) {
-          </a>
-        )}
+        <button id="discard" onClick={discard} disabled={!discard}>
+          <SvgIcon ariaHidden={true} hash="delete_forever" />
+          &nbsp;
+          {browser.i18n.getMessage("compose.discard")}
```
For consistency with the rest of the UI, let's make both of these buttons have the icon first, and then the text.

If you drop the `&nbsp;` you can use `margin-inline-end: 0.1em;` within a `.compose > .buttons > button > .icon` to apply a bit of spacing.

fezjo replied:
Ok

Standard8 reviewed (changes requested):
Thank you for the updates, I think we're almost there, just a few more tweaks.

Standard8 commented:
@fezjo did you see my comments above, would you be able to update for those? If not, would you mind if I addressed those?

I'm planning another update soon, so it would be good to get this landed.

fezjo commented:
Hello, I saw your comments but had to focus on other stuff. Sorry for not replying sooner. I can find time for this during the weekend, but I don't mind you addressing them. Thank you for your guidance, I learned a bit about UI :)

Standard8 reviewed (approved):
Thank you for all the updates, this looks good. Nice to get this fixed.
